onedal/basic_statistics/basic_statistics.py

```diff
@@ -66,17 +66,16 @@ def _get_onedal_params(self, dtype=np.float32):
             "result_option": options,
         }
 
-    def _compute_raw(self, data_table, weights_table, module, policy, dtype=np.float32):
-        params = self._get_onedal_params(dtype)
-
-        result = module.train(policy, params, data_table, weights_table)
 
-        options = self._get_result_options(self.options)
-        options = options.split("|")
+class BasicStatistics(BaseBasicStatistics):
+    """
+    Basic Statistics oneDAL implementation.
+    """
 
-        return {opt: getattr(result, opt) for opt in options}
+    def __init__(self, result_options="all", algorithm="by_default"):
+        super().__init__(result_options, algorithm)
 
-    def _compute(self, data, weights, module, queue):
+    def compute(self, data, weights=None, queue=None):
```

Comment:
```suggestion
    def fit(self, data, weights=None, queue=None):
```
Let's make it more scikit-learn-like